bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,606 @@
+                }
+            }
+        } catch (RepositoryException re) {
+            LOG.error("Repository Exception. Unable to get or create node '{}'", nextPath, re);
+            throw re;
+        }
+
+        return node;
+    }
+
+    /**
+     * Get {@link JSONObject} from URL response.
+     * @param path String
+     * @return JSONObject
+     * @throws IOException exception
+     * @throws JSONException exception
+     */
+    private JSONObject getJsonFromUri(final String path) throws IOException, JSONException {
+        try {
+            URI pathUi = new URI(null, null, path, null);
+            // we want to traverse the JCR one level at a time, hence the '1' selector.
+            URL url = new URL(this.remoteAssetsConfig.getServer() + pathUi.toString() + ".1.json");
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestProperty("Authorization", String.format("Basic %s", RemoteAssets.encodeForBasicAuth(this.remoteAssetsConfig)));
+
+            try (BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+                String line;
+                StringBuilder sb = new StringBuilder();
+
+                while ((line = br.readLine()) != null) {
+                    sb.append(line);
+                }
+
+                String sbString = sb.toString();
+
+                if (StringUtils.startsWith(sbString, "{")) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("JSON successfully fetched for URL '{}'.", url.toString());
+                    }
+                    return new JSONObject(sbString);
+                } else {
+                    LOG.error("Unable to grab JSON Object. Please ensure URL {} is valid. \nRaw Response: {}", url.toString(), sbString);
+                }
+            } finally {
+                connection.disconnect();
+            }
+        } catch (URISyntaxException e) {
+            LOG.error("URI Syntax Exception {}", e);
+        }
+
+        return null;
+    }
+
+    /**
+     * Create or update nodes from remote JSON.
+     * @param json JSONObject
+     * @param parentNode Node
+     * @throws IOException exception
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void createOrUpdateNodes(final JSONObject json, final Node parentNode) throws IOException, JSONException, RepositoryException {
+        Iterator<String> keys = json.keys();
+        while (keys.hasNext()) {
+            String key = keys.next();
+            if (json.get(key) instanceof JSONObject) {
+                if (PROTECTED_NODES.contains(key)) {
+                    continue;
+                }
+
+                String objectPath = String.format("%s/%s", parentNode.getPath(), key);
+                JSONObject objectJson = getJsonFromUri(objectPath);
+                if (objectJson == null) {
```

Comment:
Instead of handling this here, may as well handle inside `getJsonFromUri` to keep this clean here as well as the other place we call this function.